core/src/utils/transition/ios.transition.ts

```diff
@@ -123,56 +145,131 @@ const animateBackButton = (
   rootAnimation: Animation,
   rtl: boolean,
   backDirection: boolean,
-  backButtonEl: any,
-  largeTitleBox: DOMRect,
-  backButtonBox: DOMRect
+  backButtonEl: HTMLIonBackButtonElement,
+  backButtonBox: DOMRect,
+  backButtonTextEl: HTMLElement,
+  backButtonTextBox: DOMRect,
+  largeTitleEl: HTMLIonTitleElement,
+  largeTitleTextBox: DOMRect
 ) => {
   const BACK_BUTTON_START_OFFSET = rtl ? `calc(100% - ${backButtonBox.right + 4}px)` : `${backButtonBox.left - 4}px`;
-  const START_TEXT_TRANSLATE = rtl ? '7px' : '-7px';
-  const END_TEXT_TRANSLATE = rtl ? '-4px' : '4px';
-
-  const ICON_TRANSLATE = rtl ? '-4px' : '4px';
 
   const TEXT_ORIGIN_X = rtl ? 'right' : 'left';
   const ICON_ORIGIN_X = rtl ? 'left' : 'right';
 
+  const CONTAINER_ORIGIN_X = rtl ? 'right' : 'left';
+
+  /**
+   * When the title and back button texts match
+   * then they should overlap during the page transition.
+   * If the texts do not match up then the back button text scale adjusts
+   * to not perfectly match the large title text otherwise the
+   * proportions will be incorrect.
+   */
+  const doTitleAndButtonTextsMatch = backButtonTextEl.textContent === largeTitleEl.textContent;
```

Comment:
Could we `trim()` these? `textContent` includes whitespace, so an errant space at the end of one of the elements could cause wonky animation that'd be really hard to debug.

Reply:
Good catch! Fixed in https://github.com/ionic-team/ionic-framework/pull/28290/commits/4e29bffa58fb9acc29bc5acbab207c14e4487f07

Reply:
Added in https://github.com/ionic-team/ionic-framework/pull/28290/commits/f418f6b04e7b097327ef24b0b79e434eaa6e5ff4